modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/ast/ArtifactAstUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.ast;
+
+import org.mule.api.annotation.NoInstantiate;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.core.api.config.ConfigurationException;
+import org.mule.runtime.dsl.api.ConfigResource;
+import org.mule.runtime.module.artifact.activation.api.ast.AstXmlParserSupplier;
+
+import java.io.IOException;
+import java.util.Set;
+
+/**
+ * Utilities for parsing and handling {@link ArtifactAst} (for internal usage only).
+ *
+ * @since 4.5.0
+ */
+@NoInstantiate
```

Comment:
Removed the annotation, as pointed out, this class already had a private constructor to prevent instantiation, so the annotation was completely redundant.